libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2264,20 +2258,25 @@ bool TypeChecker::visit(FunctionCallOptions const& _functionCallOptions)
 		return false;
 	}
 
+	auto setCheckOption = [&](bool& _option, string const&& _name)
+	{
+		if (_option)
+			m_errorReporter.typeError(
+				_functionCallOptions.location(),
+				"Duplicate option \"" + std::move(_name) + "\"."
```

Comment:
Is "Duplicate" clear enough for the case of `f{salt: x}{salt: y}`? Would it be much work to change this (only if already set previously) to "has already been set"?